src/estimation/include/iDynTree/Estimation/ExtendedKalmanFilter.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#ifndef EXTENDED_KALMAN_FILTER_H
+#define EXTENDED_KALMAN_FILTER_H
+
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/Utils.h>
+#include <iDynTree/Core/EigenHelpers.h>
+
+namespace iDynTree
+{
+
+    /**
+     * @class DiscreteExtendedKalmanFilter naive base class implementation of discrete EKF with additive Gaussian noise
+     *
+     * @warning This class is not stand-alone but can be used meaningfully only by classes deriving from it.
+     * The system propagation function f() and measurement model function h() are virtual functions to be defined by
+     * the derived class implementing the EKF.
```

Comment:
This is cool. It probably make sense to call the class `DiscreteExtendedKalmanFilterHelper` or something similar, as in the future we may want to have a fully fledged `ExtendedKalmanFilter` that uses this class and the system representation in https://github.com/robotology/idyntree/blob/master/src/optimalcontrol/include/iDynTree/DynamicalSystem.h with the integrators in https://github.com/robotology/idyntree/tree/master/src/optimalcontrol/include/iDynTree/Integrators to have a class that can take an arbitrarly continuous system, discretize it and run the state estimation. Obviously this `ExtendedKalmanFilter` would use the  `DiscreteExtendedKalmanFilterHelper` internally.